docs/cloud/guides/security/data_masking.md

```diff
@@ -0,0 +1,177 @@
+---
+slug: /cloud/guides/data-masking
+sidebar_label: 'Data masking'
+title: 'Data masking in ClickHouse'
+description: 'A guide to data masking in ClickHouse'
+keywords: ['data masking']
+---
+
+# Data masking in ClickHouse
+
+Data masking is a technique used for data protection, in which the original data is replaced with a version of the data which maintains its format and structure while removing any personally identifiable information (PII) or sensitive information.
+This guide shows you how you can mask data in ClickHouse.
+
+## Use string replacement functions {#using-string-functions}
+
+For basic data masking use cases, the `replace` family of functions offers a convenient way to mask data:
+
+| Function                                                                                 | Description                                                                                                                                            |
+|------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [`replaceOne`](/sql-reference/functions/string-replace-functions#replaceone)             | Replaces the first occurrence of a pattern in a haystack string with the provided replacement string.                                                  |
+| [`replaceAll`](/sql-reference/functions/string-replace-functions#replaceall)             | Replaces all occurrences of a pattern in a haystack string with the provided replacement string.                                                       |
+| [`replaceRegexpOne`](/sql-reference/functions/string-replace-functions#replaceregexpone) | Replaces the first occurrence of a substring matching a regular expression pattern (in re2 syntax) in a haystack with the provided replacement string. |
+| [`replaceRegexpAll`](/sql-reference/functions/string-replace-functions#replaceregexpall) | Replaces all occurrences of a substring matching a regular expression pattern (in re2 syntax) in a haystack with the provided replacement string.      |
+
+For example, you can replace customer names with a placeholder `[CUSTOMER_NAME]` using the `replaceOne` function:
```

Comment:
Maybe be more explicit that we are replacing 'John Smith' in this example. And then perhaps add another example showing how to generalise it with the regex?